README.md

```diff
@@ -98,69 +136,103 @@ into the element. This is the only method that must be implemented by subclasses
+  * `invalidateProperty(name, oldValue)` (protected): Triggers an invalidation for
+  a specific property. This is useful when manually implementing a property setter.
+  Call `invalidateProperty` instead of `invalidate` to ensure that any configured
+  property options are honored.
+
+  * `createRenderRoot()` (protected): Implement to customize where the
   element's template is rendered by returning an element into which to
   render. By default this creates a shadowRoot for the element.
   To render into the element's childNodes, return `this`.
 
-  * `_firstRendered()`: Called after the element DOM is rendered for the first time.
-
-  * `_shouldRender(props, changedProps, prevProps)`: Implement to control if rendering
-  should occur when property values change or `invalidate` is called.
-  By default, this method always returns true, but this can be customized as
-  an optimization to avoid rendering work when changes occur which should not be rendered.
-
-  * `_render(props)`: Implement to describe the element's DOM using `lit-html`. Ideally,
-  the `_render` implementation is a pure function using only `props` to describe
-  the element template. This is the only method that must be implemented by subclasses.
-
-  * `_didRender(props, changedProps, prevProps)`: Called after element DOM has been rendered.
-  Implement to directly control rendered DOM. Typically this is not needed as `lit-html`
-  can be used in the `_render` method to set properties, attributes, and
-  event listeners. However, it is sometimes useful for calling methods on
-  rendered elements, for example focusing an input:
-  `this.shadowRoot.querySelector('input').focus()`.
-
-  * `renderComplete`: Returns a promise which resolves after the element next renders.
-
-  * `_requestRender`: Call to request the element to asynchronously re-render regardless
-  of whether or not any property changes are pending.
+## Advanced: Update Lifecycle
+
+* When the element is first connected or a property is set (e.g. `element.foo = 5`)
+  and the property's `shouldInvalidate(value, oldValue)` returns true. Then
+  * `invalidate()` tries to update the element after waiting a [microtask](https://jakearchibald.com/2015/tasks-microtasks-queues-and-schedules/) (at the end
+  of the event loop, before the next paint). Then
+    * `shouldUpdate(changedProperties)` is called and if this returns true which it
+      does by default:
+      * `update(changedProperties)` is called to update the element.
+        Note, setting properties inside `update()` will set their values but
+        will *not* trigger `invalidate()`. This calls
+        * `render()` which should return a `lit-html` TemplateResult
+          (e.g. <code>html\`Hello ${world}\`</code>)
+      * `finishFirstUpdate()` is then called to do post *first* update/render tasks.
+        Note, setting properties here will trigger `invalidate()`.
+      * `finishUpdate(changedProperties)` is then called to do post update/render tasks.
```

Comment:
finishUpdate was removed

Reply:
I found `finishUpdate` useful. What's the replacement for it now?

Reply:
Currently, it's 

```
update(changedProperties) {
  super.update(changedProperties);
  // manipulate rendered DOM.
}
```

We are considering adding `rendered` in LitElement to fulfill this need (previously `didRender`) but decided it didn't go into `UpdatingElement`.